ergotree-interpreter/src/eval/scontext.rs

```diff
@@ -23,3 +23,23 @@ pub(crate) static DATA_INPUTS_EVAL_FN: EvalFn = |_env, ctx, obj, _args| {
         elem_tpe: SType::SBox,
     }))
 };
+
+pub(crate) static SELF_BOX_INDEX_EVAL_FN: EvalFn = |_env, ctx, obj, _args| {
+    if obj != Value::Context {
+        return Err(EvalError::UnexpectedValue(format!(
+            "Context.selfBoxIndex: expected object of Value::Context, got {:?}",
+            obj
+        )));
+    }
+    let box_index = ctx
+        .ctx
+        .inputs
+        .clone()
+        .into_iter()
+        .position(|it| it == ctx.ctx.self_box)
+        .ok_or_else(|| EvalError::NotFound("Context.selfBoxIndex: box not found".to_string()));
```

Comment:
Consider
```suggestion
        .ok_or_else(|| EvalError::NotFound("Context.selfBoxIndex: box not found".to_string()))?;
    Ok(Value::Int(box_index as i32))
```
instead of `match`. 
`?` is often used to "bubble up" errors while keeping "happy path" clean. It also works if an error has a different type assuming `From` implementation is provided (compiler calls the conversion behind the scenes).

Reply:
Thanks a lot for the explanation.